### PR TITLE
[49409] Exclude work package from totals calculation by status

### PR DESCRIPTION
**Pull request by cbliard**
https://community.openproject.org/wp/49409

### CODE REVIEW

File: app/workers/work_packages/progress/apply_statuses_change_job.rb [hunk 1]
```diff
@@ -28,34 +28,40 @@
...
+  # It creates a journal entry with the System user describing the changes.
   #
   # @param status_name [String] The cause of the update to be put in the journal
   #   entry. Must be one of `VALID_CAUSE_TYPES`.
   # @param status_name [String] The name of the status to apply.
   # @param status_id [Integer] The ID of the status to apply. Not used
   #   currently, but here in case we need it in a later version.
-  # @param change [Object] The change object containing an array of [old, new]
-  #   values of the change.
+  # @param changes [Object] The changes object containing an map of array of
```
Comment (aaron-contreras) [inline app/workers/work_packages/progress/apply_statuses_change_job.rb]:
```suggestion
  # @param changes [Object] The changes object containing a map of array of
```

File: spec/support/for_each_context.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    #         let(:time) { "23:00" }
+    #       end
+    #
+    #       for_each_context "early in the morning",
+    #                        "late in the evening" do
+    #         it "has energy" do
+    #           expect(body.energy_level).to eq(100)
+    #         end
+    #       end
+    #     end
```
Comment (aaron-contreras) [inline spec/support/for_each_context.rb:58]: Very nice ⭐️

**Review by aaron-contreras — APPROVED** [review]
Brilliant ☀️

File: spec/support/for_each_context.rb (hunk 2 above)
Comment (cbliard) [reply spec/support/for_each_context.rb:58]: Thanks

**Pull request merged**